patterns/direct_processor.py

```diff
@@ -0,0 +1,142 @@
+## Goal: A simpe request pulblisher to publish a request with reply_to set to any topic.
+import os
+import platform
+import time
+
+# Import Solace Python  API modules from the solace package
+from solace.messaging.messaging_service import MessagingService, ReconnectionListener, ReconnectionAttemptListener, \
+                                                ServiceInterruptionListener, RetryStrategy, ServiceEvent
+from solace.messaging.resources.topic import Topic
+from solace.messaging.publisher.direct_message_publisher import PublishFailureListener, FailedPublishEvent
+from solace.messaging.resources.topic_subscription import TopicSubscription
+from solace.messaging.receiver.message_receiver import MessageHandler, InboundMessage
+
+
+if platform.uname().system == 'Windows': os.environ["PYTHONUNBUFFERED"] = "1" # Disable stdout buffer 
+
+# Goal is to demonstrate receive-process-reply (processor) pattern. Processor subscribes to a request topic and upon receipt of 
+# request message carrying a string, the processor converts the payload to uppercase (for simplicity) and publishes to a 
+# designated topic.
+#   direct_requestor: 
+#       subscribes to request topic 'solace/samples/python/direct/processor/input/{count}
+#       proceses the payload
+#       publishes to reply topic 'solace/samples/python/direct/processor/output'
+
+TOPIC_PREFIX = "solace/samples/python"
+
+# Broker Config. Note: Could pass other properties Look into
+broker_props = {
+    "solace.messaging.transport.host": os.environ.get('SOLACE_HOST') or "tcp://localhost:55555,tcp://localhost:55554",
+    "solace.messaging.service.vpn-name": os.environ.get('SOLACE_VPN') or "default",
+    "solace.messaging.authentication.scheme.basic.username": os.environ.get('SOLACE_USERNAME') or "default",
+    "solace.messaging.authentication.scheme.basic.password": os.environ.get('SOLACE_PASSWORD') or ""
+    }
+
+# Build A messaging service with a reconnection strategy of 20 retries over an interval of 3 seconds
+# Note: The reconnections strategy could also be configured using the broker properties object
+messaging_service = MessagingService.builder().from_properties(broker_props)\
+                        .with_reconnection_retry_strategy(RetryStrategy.parametrized_retry(20,3))\
+                        .build()
+
+# Blocking connect thread
+messaging_service.connect()
+print(f'\nMessaging Service connected? {messaging_service.is_connected}')
+
+# Inner classes for connection error handling
+class ServiceEventHandler(ReconnectionListener, ReconnectionAttemptListener, ServiceInterruptionListener):
```

Comment:
Can we move all inner classes to the top. To follow the same conventions we will have the file to start with
1. Import statements followed by
2. Environment configurations followed by
3. Constants followed by
4. Inner classes followed by
5. Main business logic followed by
6. Cleanup

Reply:
Done!